### PR TITLE
[BIOIN-1694] create docker for germline cnv tasks

### PR DESCRIPTION
**Pull request by avigailmo**

### CODE REVIEW

File: .github/actions/build-workspace-docker/action.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: "build ugbio workspace docker"
+description: "general action to build and push ugbio workspace docker"
+
+inputs:
+  workspace-folder:
+    description: "workspace folder under src"
+    required: true
+  docker-repo:
+    description: "docker image repository name"
```
Comment (inbalzelig-ugi) [inline .github/actions/build-workspace-docker/action.yml:9]: what do you mean by "docker image repository name" ?

File: .github/actions/build-workspace-docker/action.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+runs:
+  using: "composite"
+
+  steps:
+  - uses: benjlevesque/short-sha@v2.2
+    id: short-sha
+    with:
+      length: 7
+
+  - name: set tag
```
Comment (inbalzelig-ugi) [inline .github/actions/build-workspace-docker/action.yml:31]: I'm not sure about this tag logic... I think we should read it from the pyproject.toml version

File: .github/actions/build-workspace-docker/action.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      password: ${{ steps.auth.outputs.access_token }}
+
+
+  - name: Set up Docker Buildx
+    uses: docker/setup-buildx-action@v2
+
+  - name: Build and push
+    uses: docker/build-push-action@v4
+    with:
+      context: .
```
Comment (inbalzelig-ugi) [inline .github/actions/build-workspace-docker/action.yml:75]:
```suggestion
      context: ${{ inputs.docker-context }}
```

File: .github/actions/build-workspace-docker/action.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      file: ${{ inputs.docker-file }}
+      provenance: false
+      push: true
+      tags: |
+        ${{ steps.ecr-login.outputs.registry }}/${{ inputs.docker-repo }}:${{ inputs.docker-tag || steps.set-tag.outputs.imageTag }}
+        us-central1-docker.pkg.dev/ultima-data-307918/ultimagen/${{ inputs.docker-repo }}:${{ inputs.docker-tag || steps.set-tag.outputs.imageTag }}
```
Comment (inbalzelig-ugi) [inline .github/actions/build-workspace-docker/action.yml:81]: Do we always hard-coding gcp registry "us-central1-docker.pkg.dev/ultima-data-307918/ultimagen"? If the answer is yes please ignore this comment

File: .github/actions/run-workspace-tests/action.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        load: true
+        push: false
+        tags: ${{ inputs.workspace-folder }}
+        cache-from: type=gha
+        cache-to: type=gha,mode=max
+
+    - name: Run ugbio_cnv tests
+      shell: bash
+      run: |
+        docker run --rm -v ${{ github.workspace }}/src/cnv:/workdir/src/${{ inputs.workspace-folder }} ${{ inputs.workspace-folder }} pytest /workdir/src/${{ inputs.workspace-folder }}
```
Comment (inbalzelig-ugi) [inline .github/actions/run-workspace-tests/action.yml:40]:
```suggestion
        docker run --rm -v ${{ github.workspace }}/src/${{ inputs.workspace-folder }}:/workdir/src/${{ inputs.workspace-folder }} ${{ inputs.workspace-folder }} pytest /workdir/src/${{ inputs.workspace-folder }}
```

And why you have ${{ inputs.workspace-folder }} before "pytest"?

File: .github/actions/run-workspace-tests/action.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        file: ${{ inputs.docker-file }}
+        context: ${{ inputs.docker-context }}
+        provenance: false
+        load: true
+        push: false
+        tags: ${{ inputs.workspace-folder }}
+        cache-from: type=gha
+        cache-to: type=gha,mode=max
+
+    - name: Run ugbio_cnv tests
```
Comment (inbalzelig-ugi) [inline .github/actions/run-workspace-tests/action.yml:37]:
```suggestion
    - name: Run tests in the container
```

File: .github/actions/run-workspace-tests/action.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      shell: bash
+      run: |
+        docker run --rm -v ${{ github.workspace }}/src/cnv:/workdir/src/${{ inputs.workspace-folder }} ${{ inputs.workspace-folder }} pytest /workdir/src/${{ inputs.workspace-folder }}
+
+    #todo how to upload results?
```
Comment (inbalzelig-ugi) [inline .github/actions/run-workspace-tests/action.yml:42]: Why do you want to upload the results?

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -2,46 +2,19 @@ name: CI
...
-
-      - name: Set up bedops
-        run: |
-          wget https://github.com/bedops/bedops/releases/download/v2.4.41/bedops_linux_x86_64-v2.4.41.tar.bz2
-          tar -xvjf bedops_linux_x86_64-v2.4.41.tar.bz2
-          sudo mv bin/* /usr/local/bin/
 
-      - name: Install rye
-        uses: eifinger/setup-rye@v3
+      - name: Run cnv tests
```
Comment (inbalzelig-ugi) [inline .github/workflows/ci.yml:13]: Very nice and clean!

File: src/cnv/cnmops/tests/__init__.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import os
+from os.path import dirname
+
+test_dir = dirname(__file__)
+
+
+def get_resource_dir(module_file: str):
```
Comment (inbalzelig-ugi) [inline src/cnv/cnmops/tests/__init__.py:7]: I think you can make it cleaner with Path... but it is not very important

File: src/cnv/cnmops/tests/test_convert_bedGraph_to_Granges.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import filecmp
+import subprocess
+from os.path import join as pjoin
+from . import get_resource_dir
+
+resources_dir = get_resource_dir(__file__)
+script_path = "/src/cnv/cnmops/convert_bedGraph_to_Granges.R"
```
Comment (inbalzelig-ugi) [inline src/cnv/cnmops/tests/test_convert_bedGraph_to_Granges.py:7]: Is this working when running the tests inside the docker?
I thought the context in the docker is cnv.

Maybe a relative path is a better approach here

**Review by inbalzelig-ugi — CHANGES_REQUESTED** [review]
Looking good! I had a few comments/questions. My main request change is at ".github/actions/build-workspace-docker/action.yml" where you should change the context

File: .github/actions/build-workspace-docker/action.yml (hunk 1 above)
Comment (avigailmo) [reply .github/actions/build-workspace-docker/action.yml:9]: the ecr/gcr repo name without the specific image tag, like:
make_examples, trimmer, sorter etc.

File: .github/actions/build-workspace-docker/action.yml (hunk 3 above)
Comment (avigailmo) [reply .github/actions/build-workspace-docker/action.yml:75]: good catch! thanks

File: .github/actions/run-workspace-tests/action.yml (hunk 5 above)
Comment (avigailmo) [reply .github/actions/run-workspace-tests/action.yml:40]: this is how I tagged the local docker in the prev "Build docker" step 
`tags: ${{ inputs.workspace-folder }}`

**Review by inbalzelig-ugi — APPROVED** [review]

**Review by TammushB — COMMENTED** [review]
LGTM as long as all tests pass. 
also compared outputs of a full run : 
- new ug_cnmop docker : 8757895
- master: 4321453
outputs are identical.

**Review by TammushB — APPROVED** [review]

**Pull request merged**